src/shogun/classifier/svm/SVMLight.cpp

```diff
@@ -1605,14 +1605,14 @@ void SVMLight::update_linear_component_mkl(
 						W[j*num_kernels+n]+=(a[i]-a_old[i])*compute_kernel(i,j)*(float64_t)label[i];
 				}
 			}
-			w1[n]=0.0 ;
+			
```

Comment:
why do you remove this one?

Reply:
Correct me if i'm wrong, w1 vector is set to ``` [ 0 0 0 . . .  0] ``` at the start of every iteration, so I just put ```memset(w1,0,num_weights)``` at the beginning of each loop

Reply:
this only sets the nth element to 0... not the whole vector...

Reply:
True, but we already have 
```
for (int32_t i=0; i<num_kernels; i++)
		{
			w_backup[i] = old_beta[i] ;
			w1[i]=0.0 ;        // All the values are zero
		}
``` 
And then in the loop :
```
loop {
w1[n] = 1       // some nth element in num_kernel
/*code*/
w1[n] = 0     //Value is back to 0
//By the end of this iteration w1 = [ 0 0 . . ]
}
```

Reply:
indeed. but setting the whole vector 0 instead of just one element is more costly and for nothing, as the other elements are already 0 as you have pointed out above.

Reply:
Hmm. You're right, so shall I make the above-mentioned change? (w1[n] = 1, w1[n] = 0 )

Reply:
i'm not so sure why you keep trying to change the algo itself. the task is really just to replace the malloc with sgvector. you shouldn't have touched any of this code, it is 'good' as is...

the only thing that makes sense to remove is the initial zeroing of the vector as the SGVector constructor takes care of that.

Reply:
Sorry I thought maybe we could optimize the algorithm, haha I got too excited !